python/ray/util/spark/databricks_hook.py

```diff
@@ -89,9 +104,14 @@ def on_ray_dashboard_created(self, port):
     def on_cluster_created(self, ray_cluster_handler):
         db_api_entry = _get_db_api_entry()
         try:
-            db_api_entry.registerBackgroundSparkJobGroup(
-                ray_cluster_handler.spark_job_group_id
-            )
+            # We only cancel spark job group when global mode is disabled,
+            # otherwise even when the parent REPL is detached or died, we
+            # keep the spark job group alive so that ray cluster is alive
+            # and can be connected again.
+            if not global_mode_enabled():
+                db_api_entry.registerBackgroundSparkJobGroup(
+                    ray_cluster_handler.spark_job_group_id
+                )
         except Exception:
             _logger.warning(
                 "Registering Ray cluster spark job as background job failed. "
```

Comment:
If global mode is enabled, we need to set `auto_shutdown_minutes` to 0 to disable auto shutdown. @serena-ruan

Reply:
Why? They can still change the auto_shutdown_minutes if they want to keep it running right?

Reply:
Because multiple databricks notebooks might connect to one global mode ray cluster,
see logic in `auto_shutdown_watcher`, in this case, every notebook starts a  `auto_shutdown_watcher`, if anyone of these notebooks become inactive for "auto_shutdown_minutes" duration, then it triggers ray cluster shutdown, but other notebooks might still using the ray cluster.

so auto_shutdown_watcher implementation is incompatible with global mode ray cluster.

Reply:
Isn't auto_shutdown_watcher only triggered once when we create the global ray cluster? It's inside `on_cluster_created` 🤔

Reply:
oh, you are right. auto_shutdown_watcher only triggered once when we create the global ray cluster.

But there is still an issue: once the notebook that creates the ray cluster is detached, the auto_shutdown_watcher is also terminated. Then auto_shutdown does not work any more.